src/kaleidoscope/plugin/IdleLEDs.cpp

```diff
@@ -25,23 +26,33 @@ uint16_t IdleLEDs::idle_time_limit = 600; // 10 minutes
 uint32_t IdleLEDs::end_time_;
 
 EventHandlerResult IdleLEDs::beforeEachCycle() {
-  if (!::LEDControl.paused &&
-      (Kaleidoscope.millisAtCycleStart() >= end_time_)) {
+  if (Kaleidoscope.millisAtCycleStart() < end_time_)
+    return EventHandlerResult::OK;
+
+  if (::Kaleidoscope.has_leds && !::LEDControl.paused) {
     ::LEDControl.set_all_leds_to(CRGB(0, 0, 0));
     ::LEDControl.syncLeds();
 
     ::LEDControl.paused = true;
   }
+  if (::UnderglowControl.has_leds() && !::UnderglowControl.isPaused()) {
+    ::UnderglowControl.setColor(0, 0, 0);
+    ::UnderglowControl.sync();
+    ::UnderglowControl.pause();
```

Comment:
LEDControl and this should have parity. If this is going to be a method here (which I like), it should get added to LEDControl at the same time.

Reply:
Ok. I didn't want to touch `LEDControl` yet, before we agree on naming & API, happy to add the new methods there!